src/models/hooks/goal.js

```diff
@@ -62,6 +63,16 @@ const preventNameChangeWhenOnApprovedAR = (_sequelize, instance) => {
   }
 };
 
+const onlyAllowTrGoalSourceForGoalsCreatedViaTr = (_sequelize, instance) => {
```

Comment:
this code is identical to the one in activityReportGoal (right?) Can we please extract to a helper file to avoid repeats?

Reply:
sure